contracts/v0.6/SimpleSwap.sol

```diff
@@ -0,0 +1,178 @@
+pragma solidity ^0.6.0;
+
+import "@chainlink/contracts/src/v0.6/Owned.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "./token/ERC677Receiver.sol";
+
+contract SimpleSwap is Owned, ReentrancyGuard {
+  using SafeMath for uint256;
+
+  event LiquidityUpdated(
+    uint256 indexed amount,
+    address indexed source,
+    address indexed target
+  );
+  event TokensSwapped(
+    uint256 indexed amount,
+    address indexed source,
+    address indexed target,
+    address caller
+  );
+
+  mapping(address => mapping(address => uint256)) private s_swappableAmount;
+
```

Comment:
Maybe we could disallow Ether transfers, to avoid accidentally locked ETH:

```suggestion

/**
 * @dev Disallows direct send by setting a default function without the `payable` flag.
 */
function() external {}

```

We could also check that no ETH is sent via the public interface:

```solidity
require(msg.value == 0, "ETH transfers not allowed");
```